app/lib/differential_expression_service.rb

```diff
@@ -200,6 +200,13 @@ def self.validate_annotation(cluster_file, study, annotation_name, annotation_ty
     identifier = "#{annotation_name}--#{annotation_type}--#{annotation_scope}"
     raise ArgumentError, "#{identifier} is not present" if annotation.nil?
     raise ArgumentError, "#{identifier} cannot be visualized" unless can_visualize
+
+    # last, validate that the requested annotation & cluster will provide a valid intersection of annotation values
+    # specifically, discard any annotation/cluster combos that only result in one distinct label
+    cells_by_label = ClusterVizService.cells_by_annotation_label(cluster, annotation_name, annotation_type, annotation_scope)
```

Comment:
Blocker: let's start addressing the extraneous `annotation_type` parameter.  I don't object to further refining it later, but I think it's important we start.

```suggestion
    cells_by_label = ClusterVizService.cells_by_annotation_label(cluster, annotation_name, 'group', annotation_scope)
```